base/bucket_gocb.go

```diff
@@ -2112,6 +2113,110 @@ func (bucket CouchbaseBucketGoCB) Close() {
 	}
 }
 
+// This flushes the bucket but doesn't actually close the bucket, despite the name CloseAndDelete(),
+// which is based on the sgbucket.DeleteableBucket interface.
+func (bucket CouchbaseBucketGoCB) CloseAndDelete() error {
+
+	bucketManager, err := bucket.getBucketManager()
+	if err != nil {
+		return err
+	}
+
+	workerFlush := func() (shouldRetry bool, err error, value interface{}) {
+		err = bucketManager.Flush()
+		if err != nil {
+			Warn("Error flushing bucket: %v  Will retry.", err)
+		}
+		shouldRetry = (err != nil) // retry (until max attempts) if there was an error
```

Comment:
Could move up into the block above and simplify: `shouldRetry = true`

Reply:
Good catch